zenoh-plugin-ros2dds/src/lib.rs

```diff
@@ -245,27 +300,42 @@ pub async fn run(runtime: Runtime, config: Config) {
         }
     };
 
-    // if "ros_localhost_only" is set, configure CycloneDDS to use only localhost interface
-    if config.ros_localhost_only {
-        if ros_distro_is_less_than("iron") {
+    // Dynamic Discovery is changed after iron. Need to check the ROS 2 version.
+    // https://docs.ros.org/en/rolling/Tutorials/Advanced/Improved-Dynamic-Discovery.html
+    if ros_distro_is_less_than("iron") {
+        // if "ros_localhost_only" is set, configure CycloneDDS to use only localhost interface
```

Comment:
To avoid confusion, I think here we should check if `config.ros_automatic_discovery_range` or `config.ros_static_peers` are different than `None` and log a warning explaining they will be ignored because it's not supported before Iron.

Reply:
Make sense